### ISSUE TITLE
TensorFlow compatibility problems

### ISSUE THREAD

**Issue by TITAN-PyCompat** [omoindrot/tensorflow-triplet-loss#42]
Dear developers,

I observed some TensorFlow API usage in your code that might lead to serious compatibility problems(like a crash).

For tf API `tf.reduce_max` and `tf.reduce_min`, their parameter `keepdims` appears only since tf v1.5.0-rc0. If use this keyword to pass the value, the program will crash in older versions of TensorFlow. It's better to use `keep_dims` for earlier versions.

I have made a patch for this problem for you in #41 
Look forward to your response! @ymcasky @dlfelps @omoindrot 
Thanks!
